docs/samples/python/transformers_recognizer/transformer_recognizer.py

```diff
@@ -0,0 +1,289 @@
+        device = 0 if torch.cuda.is_available() else -1
+        self.pipeline = pipeline(
+            "ner",
+            model=AutoModelForTokenClassification.from_pretrained(
+                self.model_path),
+            tokenizer=AutoTokenizer.from_pretrained(self.model_path),
+            # Will attempt to group sub-entities to word level
+            aggregation_strategy=self.aggregation_mechanism,
+            device=device,
+            framework='pt',
+            ignore_labels=self.ignore_labels,
+
+        )
+
+        self.is_loaded = True
+
+    def get_supported_entities(self) -> List[str]:
+        """
+        Return supported entities by this model.
+        :return: List of the supported entities.
+        """
+        return self.supported_entities
+
+    # Class to use transformers with Presidio as an external recognizer.
+    def analyze(
+        self, text: str, entities: List[str], nlp_artifacts: NlpArtifacts = None
+    ) -> List[RecognizerResult]:
+        """
+        Analyze text using transformers model to produce NER tagging.
+        :param text(str) : The text for analysis.
+        :param entities(List[str]): Not working properly for this recognizer.
+        :param nlp_artifacts: Not used by this recognizer.
+        :return: The list of Presidio RecognizerResult constructed from the recognized
+            transformers detections.
+        """
+
+        results = list()
+        # Run transformer model on the provided text
+        ner_results = self._get_ner_results_for_text(text)
+
+        for res in ner_results:
+            res['entity_group'] = self.__check_label_transformer(
+                res["entity_group"])
+            textual_explanation = self.default_explanation.format(
+                res["entity_group"]
+            )
+            explanation = self.build_transformers_explanation(
+                round(res["score"], 2), textual_explanation, res["word"]
+            )
+            transformers_result = self._convert_to_recognizer_result(
+                res, explanation
+            )
+
+            results.append(transformers_result)
+
+        return results
+
+    @ staticmethod
+    def split_text_to_word_chunks(input_length: int, chunk_length: int, overlap_length: int) -> List[List]:
+        """The function calculates chunks of text with size chunk_length. Each chunk has overlap_length number of
+        words to create context and continuity for the model
+
+        Args:
+            input_length (int): Length of input_ids for a given text
+            chunk_length (int): Length of each chunk of input_ids. Should match the max input length of the transformer model
+            overlap_length (int): Number of overlapping words in each chunk
+
+        Returns:
+            List[List]: List of start and end position for each text chunk
+        """
+        return [[i, min([i+chunk_length, input_length])] for i in range(0, input_length, chunk_length-overlap_length)]
```

Comment:
Will this create overlapping chunks at the end of the text? For example, for:
```python
input_length=100
chunk_length=15
overlap_length=10
```
We would get
```
[[0, 15],
 [5, 20],
 [10, 25],
 [15, 30],
 [20, 35],
 [25, 40],
 [30, 45],
 [35, 50],
 [40, 55],
 [45, 60],
 [50, 65],
 [55, 70],
 [60, 75],
 [65, 80],
 [70, 85],
 [75, 90],
 [80, 95],
 [85, 100],
 [90, 100],
 [95, 100]]
```

Maybe we can iterate only until `input_length-overlap_length`. WDYT?

Reply:
Should remove some of the duplicate windows in the end, great catch. 
In general, this feature should be redesigned as at the moment it splits text in the middle of a word, which could create potential errors in the future.